### PR TITLE
Deprecating the Rinkeby, Ropsten, and Kovan test networks

### LINKED ISSUE #15957: Deprecate Rinkeby, Ropsten and Kovan
This is a follow up from https://github.com/MetaMask/metamask-extension/issues/15295.
The time has come for us to say goodbye to some of our beloved test networks 💔. [Infura will deprecate support to Rinkeby, Ropsten and Kovan on October 5th](https://blog.infura.io/post/deprecation-timeline-for-rinkeby-ropsten-and-kovan-testnets) so we should remove those networks from our default list around the same time. 

## Acceptance Criteria
1. Rinkeby, Ropsten and Kovan are not on MM default network lists anymore;
2. Users are able to add Rinkeby, Ropsten and Kovan to their MM through the custom network form;
3. Users are able to add Rinkeby, Ropsten and Kovan to their MM through the dapp API;
4. This is released in v10.20 or ahead of that in a hot fix release to match the timing with Infura's deprecation.

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/MetaMask/metamask-extension/issues/15957

## Explanation
These testnets are being deprecated and therefore will no longer be in our default network list.

Goerli will now take the place of Rinkeby as the default network when in debug.

## Manual Testing Steps

1. Rinkeby, Ropsten and Kovan are not on MM default network lists anymore
2. Users are able to add Rinkeby, Ropsten and Kovan to their MM through the custom network form
3. Users are able to add Rinkeby, Ropsten and Kovan to their MM through the dapp API (Test using chainlist)
4. Network is set to Goerli when starting fresh in debug mode.

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/components/ui/deprecated-test-networks/deprecated-test-networks.js [hunk 1]
```diff
@@ -21,12 +21,7 @@ export default function DeprecatedTestNetworks() {
   const t = useI18nContext();
 
   useEffect(() => {
-    if (
```
Comment (danjm) [inline ui/components/ui/deprecated-test-networks/deprecated-test-networks.js:24]: Why remove `0x3` and `0x2a` as networks on which this warning will be shown?

**Comment by danjm** [comment]
@ryanml I think we need to add a migration such that if a user is on one of these networks then the data for that network is added as a custom network. Otherwise, once the infura support is turned off, the network they are on will just be broken... possibly causing errors that result in crashes

**Comment by danjm** [comment]
@ryanml I added a migration to meet the requirements discussed here https://consensys.slack.com/archives/G1L7H42BT/p1664365963983489?thread_ts=1664206906.853299&cid=G1L7H42BT

**Comment by danjm** [comment]
When I ran this locally (before adding my migration), if I already had an install of metamask and rinkeby was selected, after updating the build to include these changes, there was an error in the network controller... something about not recognizing a network... might have been an issue with local data and setup, but worth manually testing this case @ryanml

**Comment by seaona** [comment]
As @danjm is mentioning, I'm also getting some trouble upgrading from another version, having Rinkeby selected, and then loading this version.

I am seeing the Restart MetaMask page and this error on the background console:

image

Here the flow:
[testnets-deprecation.webm](https://user-images.githubusercontent.com/54408225/192811388-c657c82d-e1d8-4788-b83c-6f8dc10c08e4.webm)

**Comment by danjm** [comment]
I think I have a fix to the error @seaona  and I mentioned above. I will confirm locally and then push

**Comment by danjm** [comment]
@seaona @ryanml the above bug should be fixed with the latest commit

**Comment by seaona** [comment]
If these testnets are added manually, should they be placed in the `Test networks` section?

image

**Comment by seaona** [comment]
> @seaona @ryanml the above bug should be fixed with the latest commit

Thank you! I see it fixed now :ok_hand:

**Comment by danjm** [comment]
I have also now added code for these two requirements:

2. When on those custom networks, if the rpc url is an infura one, continue to show the existing warning.
3. When on those custom networks, if the rpc url is an infura one and we’re having issue connecting to it, add a warning to network network loading overlay “The current rpc url for this network has been deprecated. Click here to set a new one.” where here is a link that takes the user to the edit network page.


https://user-images.githubusercontent.com/7499938/192831124-55a7839d-1e54-4209-8bd6-d8987102ad74.mp4

**Review by danjm — DISMISSED** [review]

**Review by ryanml — COMMENTED** [review]
Can’t approve my own PR but @danjm your code changes look good, thanks!

**Review by digiwand — DISMISSED** [review]
Nice work! Tested w/ Chrome x macOS

tiny, non-blocking note: I think we can remove the `mayBeFauceting` comment from `scripts/metamask-controller.js` with Ropsten deprecated

-----

so-long-partner

**Comment by ryanml** [comment]
I've fixed a merge conflict so this will need a re-approval, thanks! cc: @danjm @digiwand

**Review by danjm — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**